app/models/agents/digest_agent.rb

```diff
@@ -24,12 +28,19 @@ class DigestAgent < Agent
     def default_options
       {
           "expected_receive_period_in_days" => "2",
-          "message" => "{{ events | map: 'message' | join: ',' }}"
+          "message" => "{{ events | map: 'message' | join: ',' }}",
+          "retained_events" => "0"
       }
     end
 
     form_configurable :message, type: :text
     form_configurable :expected_receive_period_in_days
+    form_configurable :retained_events
+
+    def validate_options
+      errors.add(:base, 'retained_events must be 0 to 999') unless options['retained_events'].to_i >= 0 && options['retained_events'].to_i < 1000
+      errors.add(:base, 'retained_events must be an integer') unless (Integer(options['retained_events']).is_a? Integer rescue false)
```

Comment:
I think we need to remove that validation, existing DigestAgents do not have `retained_events` set and it will cause the validation to fail (the same reason the specs are not passing).

You are casting the value using `to_i` which defaults to 0 when passed `nil` or an empty string, so it should be good without explicitly validating it.